Thank you ChatGPT. I was manually stepping though the setup_wyoming_voice_ai.sh file to try to find out where the error is.

I ran: nickspi5@raspberrypi1:~/Downloads/Academy-main/Lesson6 $ pip3 install --upgrade pip
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
nickspi5@raspberrypi1:~/Downloads/Academy-main/Lesson6 $ 

I previously created a virtual environment named venv.

Should I open the virtual environment first before running pip3 install --upgrade pip or is there another way to resolve this error?

